### ISSUE TITLE
FFmpegFrameGrabber open rtsp stream , sometimes ffmpegFrameGrabber.grab()  will be interrupted , so my camera view pause

### ISSUE THREAD

**Issue by MrQ-Android** [bytedeco/javacv#1256]
FFmpegFrameGrabber open rtsp stream , sometimes ffmpegFrameGrabber.grab()  will be interrupted , so my camera view pause

my JavaCV version 1.4.4

**Comment by saudet** [comment]
Yes, that's normal when network connections fail. What is the issue?

**Comment by MrQ-Android** [comment]
Thank you very much for your ask! 
when the grab() interrupted (if the network reason) , How can i resume grab() and  resume the preview?

**Comment by saudet** [comment]
We can call `restart()` for that.

**Comment by MrQ-Android** [comment]
ok I know !
I have another question, grab interrupted the thread  or  grab result be null ?

this is my code : 

isReceiving = true;
        receiveTask = executor.submit(() -> {
            while (isReceiving && !isRelease) {

                try {
                    //获取Mat
                    mat = CameraUtil.frame2Mat(ffmpegFrameGrabber.grab());
                    if (mat == null) {
                        continue;
                    }

                    //Mat --> image
                    if (previewCallback != null) {
                        final Image img = CameraUtil.mat2Image(mat);
                        if (img != null) {
                            previewCallback.onPreview(img);
                        }
                    }

                    //mat -> byte
                    byte[] data = CameraUtil.mat2Bytes(mat);

                    //track
                    offer(data);


                } catch (FrameGrabber.Exception e) {
                    e.printStackTrace();
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }

            if (cameraStatusCallback != null) {
                cameraStatusCallback.offline(getCameraId(), System.currentTimeMillis());
            }
        });

**Comment by MrQ-Android** [comment]
ok I know !
I have another question, grab interrupted the thread or grab result be null ?

this is my code :

    isReceiving = true;
    receiveTask = executor.submit(() -> {
    while (isReceiving && !isRelease) {

            try {
                //获取Mat
                mat = CameraUtil.frame2Mat(ffmpegFrameGrabber.grab());
                if (mat == null) {
                    continue;
                }

                //Mat --> image
                if (previewCallback != null) {
                    final Image img = CameraUtil.mat2Image(mat);
                    if (img != null) {
                        previewCallback.onPreview(img);
                    }
                }

                //mat -> byte
                byte[] data = CameraUtil.mat2Bytes(mat);

                //track
                offer(data);


            } catch (FrameGrabber.Exception e) {
                e.printStackTrace();
            } catch (Exception e) {
                e.printStackTrace();
            }
        }

        if (cameraStatusCallback != null) {
            cameraStatusCallback.offline(getCameraId(), System.currentTimeMillis());
        }
    });

**Comment by saudet** [comment]
We can get any sort of errors on failure. If this happens often and you need to recover automatically, you'll need to do a lot of testing and make sure to cover all the relevant situations in your case.

**Comment by MrQ-Android** [comment]
ok  Thank you  ! best wishes

**Comment by saudet** [comment]
You'll also want to set timeout values as demonstrated in this sample code:
https://github.com/bytedeco/javacv/blob/master/samples/FFmpegStreamingTimeout.java

**Comment by MrQ-Android** [comment]
Thank u
